### DISCUSSION TITLE
Make `transpileModules` accepting a regexp or a path glob

### DISCUSSION THREAD

**Discussion by just-boris** [vercel/next.js#51293] [Ideas]
### Goals

Currently `transpileModules` configuration requires listing packages one-by-one, including transitive dependencies. However, this does not scale, because may change often and this requires manual step when upgrading dependencies.

For example when [using Cloudscape design components with Next.js](https://cloudscape.design/get-started/integration/using-cloudscape-components/#bundling-with-next.js-optional), you need to have this config

```js
module.exports = {
  transpilePackages: [
    '@cloudscape-design/components',
    '@cloudscape-design/components-toolkit'
  ]
};
```

In the future, `@cloudscape-design/components` can get more dependencies which will require a follow up configuration change. 

It would be great, if `transpilePackages` would support a way to enable transpilation for the entire scope (because modules from a single scope typically follow consistent conventions)

```js
module.exports = {
  transpilePackages: ['@cloudscape-design/**']
};
```

### Non-Goals

n/a

### Background

We have encountered similar issues with Jest testing framework, so we created [a preset](https://github.com/cloudscape-design/jest-preset/blob/main/jest-preset.js#L14) which enables transpilation for the entire `node_modules/@cloudscape-design` scope

### Proposal

Not very familiar with codebase, but I can try implementing this

**Comment by wottpal** [comment]
Would love to see this. Anyone in the meantime found a workaround?
